plugins/source/hubspot/client/client.go

```diff
@@ -0,0 +1,77 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/clarkmcc/go-hubspot"
+	"github.com/cloudquery/plugin-sdk/plugins/source"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/specs"
+	"github.com/rs/zerolog"
+)
+
+// Empirically tested that this is the largest page size that HubSpot allows.
+const DefaultPageSize = 100
+
+type Client struct {
+	Authorizer *hubspot.TokenAuthorizer
+
+	// Used for fetching associations.
+	// We use the `v3` library (since the `clakmcc/go-hubspot` doesn't support `v4` yet).
+	// The list was taken from https://developers.hubspot.com/docs/api/crm/associations/v3
+	AllObjectTypes []string
+
+	// Used for multiplexing when fetching `crm_pipelines`.
+	ObjectType string
+
+	Logger zerolog.Logger
+}
+
+func (c *Client) ID() string {
+	if c.ObjectType != "" {
+		return "hubspot:" + c.ObjectType
+	}
+	return "hubspot"
+}
+
+// Used for multiplexing when fetching `crm_pipelines`
+func (c *Client) withObjectType(objectType string) *Client {
+	newClient := *c
+	newClient.Logger = c.Logger.With().Str("object_type", objectType).Logger()
+	newClient.ObjectType = objectType
+	return &newClient
+}
+
+func New(ctx context.Context, logger zerolog.Logger, s specs.Source, _ ...source.Option) (schema.ClientMeta, error) {
+	var pluginSpec Spec
+
+	if err := s.UnmarshalSpec(&pluginSpec); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal plugin spec: %w", err)
+	}
+
+	authToken := os.Getenv("HUBSPOT_APP_TOKEN")
+	if authToken == "" {
+		return nil, fmt.Errorf("failed to get hubspot auth token. Please provide an auth-token (see https://www.cloudquery.io/docs/plugins/sources/hubspot/overview#authentication)")
+	}
+
+	objectTypes := []string{
+		"Contacts",
+		"Companies",
+		"Deals",
+		"Tickets",
+		"Calls",
+		"Quotes",
+		"Line_items",
+		"Meetings",
+		"Products",
+		"Feedback_submissions",
+	}
+
+	return &Client{
+		Logger:         logger,
+		Authorizer:     hubspot.NewTokenAuthorizer(authToken),
+		AllObjectTypes: objectTypes,
```

Comment:
Is `AllObjectTypes` used anywhere? I couldn't find where

Reply:
Yep, leftovers from deleted code :)